```
[ERROR] Tests run: 6, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 23.773 s <<< FAILURE! - in org.apache.hadoop.hdfs.TestDatanodeRegistration
[ERROR] testForcedRegistration(org.apache.hadoop.hdfs.TestDatanodeRegistration)  Time elapsed: 11.562 s  <<< FAILURE!
java.lang.AssertionError
    at org.junit.Assert.fail(Assert.java:87)
    at org.junit.Assert.assertTrue(Assert.java:42)
    at org.junit.Assert.assertTrue(Assert.java:53)
    at org.apache.hadoop.hdfs.TestDatanodeRegistration.testForcedRegistration(TestDatanodeRegistration.java:385)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
```

Unclear if this is a flaky test: https://issues.apache.org/jira/browse/HDFS-15646

Heres the failed assertion: https://github.com/apache/hadoop/blob/03cfc852791c14fad39db4e5b14104a276c08e59/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDatanodeRegistration.java#L385

--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminManager.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeAdminManager.java
@@ -636,7 +636,6 @@ public class DatanodeAdminManager {
               + "{}.", dn, e);
           pendingNodes.add(dn);
           toRemove.add(dn);
-          unhealthyDns.remove(dn);
         } finally {
           iterkey = dn;
         }

--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDecommission.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestDecommission.java
@@ -1657,7 +1657,7 @@ public class TestDecommission extends AdminStatesBaseTest {
     GenericTestUtils.waitFor(() -> decomManager.getNumTrackedNodes() == numDeadNodes
             && decomManager.getNumPendingNodes() == numLiveDecommNodes
             && liveDecommNodes.stream().allMatch(
-        node -> node.getAdminState().equals(AdminStates.DECOMMISSION_INPROGRESS)),
+                node -> node.getAdminState().equals(AdminStates.DECOMMISSION_INPROGRESS)),
         500, 30000);
     assertThat(liveDecommNodes)
         .as("Check all live decommissioning nodes queued in DatanodeAdminManager")
@@ -1722,7 +1722,7 @@ public class TestDecommission extends AdminStatesBaseTest {
       }
       return decomManager.getNumTrackedNodes() == 0 && decomManager.getNumPendingNodes() == 0
           && deadNodeProps.keySet().stream().allMatch(
-          node -> node.getAdminState().equals(AdminStates.DECOMMISSIONED));
+              node -> node.getAdminState().equals(AdminStates.DECOMMISSIONED));
     }, 500, 30000);
   }
 

